packages/vm/src/instance.rs

```diff
@@ -91,21 +82,10 @@ where
         module: &Module,
         backend: Backend<A, S, Q>,
         gas_limit: u64,
-        print_debug: bool,
         extra_imports: Option<HashMap<&str, Exports>>,
         instantiation_lock: Option<&Mutex<()>>,
     ) -> VmResult<Self> {
-        let fe = FunctionEnv::new(&mut store, {
-            let e = Environment::new(backend.api, gas_limit);
-            if print_debug {
-                e.set_debug_handler(Some(Rc::new(RefCell::new(
-                    |msg: &str, _info: DebugInfo<'_>| {
-                        eprintln!("{msg}");
-                    },
-                ))))
-            }
-            e
-        });
+        let fe = FunctionEnv::new(&mut store, Environment::new(backend.api, gas_limit));
```

Comment:
Looks good. A caller can do this now quite easily and powerful: https://github.com/CosmWasm/cosmwasm/blob/385efdb606df77bdeeae15966dffee9759436d91/packages/vm/examples/memory.rs#L139-L143